### PR TITLE
Initial Plugin

### PR DESCRIPTION
**Pull request by 5c077yP**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,78 @@
...
+server.route({
+  method: 'GET',
+  path: '/test',
+  handler(request, reply) {
+    return reply('test handler called');
+  },
+  config: {
+    plugins: {
+      'hapi-demo': {
+        test(request) {
```
Comment (alexbeletsky) [inline README.md:49]: `test`.. maybe somethins react style `shouldApplyRoute()` or something?

File: README.md [hunk 2]
```diff
@@ -1 +1,78 @@
...
+      })
+    }
+  }
+});
+```
+
+Now when calling the `/test` route without any query parameters the normal hander will be called, but when
+calling it with query params like `/test?demo=true`, then the demo handler will be called;
+
+### Api
```
Comment (alexbeletsky) [inline README.md:69]: API

File: README.md (hunk 1 above)
Comment (alexbeletsky) [inline README.md:49]: also, I think it's better to not only have `request` as parameter, but also an options table. This plug could be theoretically used to put API's / Apps to maintenance mode, with one toggle 

https://github.com/likeastore/maintenance

Comment (5c077yP) [inline README.md:49]: @alexbeletsky i think passing the options is not really necessary as the request object contains the server object, which itself then contains the options

**Pull request closed without merging**